### PR TITLE
Add Cocina ActiveRecord models.

### LINKED ISSUE #3138: Implement new object models and object store
* Implement new AdminPolicy, Collection, and DRO models as shown in https://github.com/sul-dlss/dor-services-app/pull/2413
* Implement object store (called `CocinaObjectStore`) that provides an abstraction around these models as shown in https://github.com/sul-dlss/dor-services-app/blob/a181ada6db35151a26b3acc4ace207198ef988cb/app/services/cocina/object_store.rb. (That is, it allows finding, deleting, and saving cocina models without knowing about the underlying ActiveRecord models.)

Note that this does not include adding versioning.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #3138

## Why was this change made?
~~Fedora~~

## How was this change tested?
Unit

### CODE REVIEW

File: db/migrate/20220131194912_create_collection.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class CreateCollection < ActiveRecord::Migration[5.2]
+  def change
+    create_table :collections do |t|
+      t.string :external_identifier, null: false
+      t.string :cocina_version, null: false
+      # Type is reserved by Rails.
+      t.string :content_type, null: false
```
Comment (jcoyne) [inline db/migrate/20220131194912_create_collection.rb:7]: Why is content_type necessary for collections? Isn't this constant?  If not, can this be changed to collection_type as content_type is an attribute that has specific meaning in sdr.

File: spec/factories/admin_policies.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :admin_policy do
+    cocina_version { '0.0.1' }
```
Comment (jcoyne) [inline spec/factories/admin_policies.rb:5]: Should this be Cocina::Models::VERSION?

File: spec/factories/collections.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :collection do
+    cocina_version { '0.0.1' }
```
Comment (jcoyne) [inline spec/factories/collections.rb:5]: Cocina::Models::VERSION?

File: spec/factories/dros.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :dro do
+    cocina_version { '0.0.1' }
```
Comment (jcoyne) [inline spec/factories/dros.rb:5]: Cocina::Models::VERSION?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: db/migrate/20220131194912_create_collection.rb (hunk 1 above)
Comment (justinlittman) [reply db/migrate/20220131194912_create_collection.rb:7]: Not a constant: https://github.com/sul-dlss/cocina-models/blob/main/openapi.yml#L394-L399

I will rename to `collection_type`.

File: spec/factories/admin_policies.rb (hunk 2 above)
Comment (justinlittman) [reply spec/factories/admin_policies.rb:5]: Could be, but doesn't have to be.

My general strategy for test fixtures is to use `Cocina::Models::VERSION` when the version matters and `0.0.1` when it doesn't. However, if you feel strongly happy to change in these tests.

**Review by jcoyne — APPROVED** [review]

File: app/models/admin_policy.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# Model for an AdminPolicy.
+class AdminPolicy < ApplicationRecord
+  # @return [Cocina::Models::AdminPolicy] Cocina Administrative Policy
+  def to_cocina
+    Cocina::Models::AdminPolicy.new({
+      cocinaVersion: cocina_version,
+      type: Cocina::Models::Vocab.adminPolicy,
```
[the remark this answers is not in the document]
Comment (justinlittman) [reply app/models/admin_policy.rb:9]: Fixing.

**Pull request merged**